Solutions/GoogleWorkspaceReports/Data Connectors/GWorkspaceReportsAPISentinelConnector/__init__.py

```diff
@@ -158,20 +212,82 @@ def gen_chunks_to_object(data,chunksize=100):
     yield chunk
 
 def gen_chunks(data,log_type):
-    for chunk in gen_chunks_to_object(data, chunksize=2000):
-        body = json.dumps(chunk)
-        post_data(customer_id, shared_key,body,log_type)
+    #for chunk in gen_chunks_to_object(data, chunksize=2000):
+        #body = json.dumps(chunk)
+        #post_data(customer_id, shared_key,body,log_type)
+    chunks = [data[i:i+chunksize] for i in range(0, len(data), chunksize)]
+    logging.info("Entered into the chunks mode")
+    latest_timestamp = "";
+    i = 0
+    for chunk in chunks:
+        try:
+            i = i+1
+            logging.debug("Iteration chunk {}".format(i))
+            body = json.dumps(chunk)
+            logging.debug(body)
```

Comment:
Size can be calculated here and make sure lesser than 25 MB before posting to LA

Reply:
handled this before posting it to LA in recent logic